processing/src/main/java/org/apache/druid/query/rowsandcols/ArrayListRowsAndColumns.java

```diff
@@ -153,11 +154,28 @@ public Column findColumn(String name)
       return new LimitedColumn(retVal, startOffset, endOffset);
     }
 
-    final Function<RowType, Object> adapterForValue = rowAdapter.columnFunction(name);
     final Optional<ColumnType> maybeColumnType = rowSignature.getColumnType(name);
     final ColumnType columnType = maybeColumnType.orElse(ColumnType.UNKNOWN_COMPLEX);
     final Comparator<Object> comparator = Comparator.nullsFirst(columnType.getStrategy());
 
+    final Function<RowType, Object> adapterForValue;
+    if (columnType.equals(ColumnType.STRING)) {
+      // special handling to reject MVDs
+      adapterForValue = f -> {
+        Object value = rowAdapter.columnFunction(name).apply(f);
+        if (value instanceof Object[] || value instanceof List) {
```

Comment:
note: isn't there a method somewhere which have a distinctive name ; and could decide if the value is an MVD or not; going down to `instanceof` feels like going down a few abstraction levels

Reply:
an mvd should never show up as `Object[]`, they are always lists. Part of the problem is that mvds do not have a distinctive column type, you can only tell from `ColumnCapabilities`, but these are for the most part only accurate when fetched from physical segments, and typically not accurate/reconstructed from a `ColumnType` on the broker (and afaik unavailable to `RowsAndColumns`)

Reply:
Updated it to just `value instanceof List`